frontend/src/stores/filterState/FilterStateStore.ts

```diff
@@ -1,76 +1,97 @@
 /* istanbul ignore file */
 import { injectable } from 'inversify';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 import SimpleStore from '../SimpleStore';
-import {
-  FilterLineState,
-  FilterPropertyState,
-  FilterState,
-} from './FilterState';
+import { FilterPropertyState, FilterState } from './FilterState';
 
 /**
  * Contains the current state of the filter.
  */
 @injectable()
 export default class FilterStateStore extends SimpleStore<FilterState> {
   protected getInitialValue(): FilterState {
-    return { edges: [], nodes: [] };
+    return new FilterState([], []);
+  }
+
+  public getValue(): FilterState {
+    const json = super.getValue();
+    return new FilterState(json.edges, json.nodes);
+  }
+
+  public getState(): Observable<FilterState> {
+    return super
+      .getState()
+      .pipe(map((next) => new FilterState(next.edges, next.nodes)));
   }
 
   /**
-   * Toggles the {@link isActive} property of the specified filter line.
+   * Toggles the {@link isActive} property of the specified filter line of the store value.
+   * Updates the state of the store afterwards.
    * @param filterLineType - filter line which {@link isActive} property is toggled
    * @param entity - specifies whether this entity is a node or an edge
    */
   public toggleFilterLineActive(
     filterLineType: string,
     entity: 'node' | 'edge'
   ): void {
-    const line: FilterLineState | undefined = this.getLineFromFilterState(
-      filterLineType,
-      entity
-    );
+    this.getValue().toggleFilterLineActive(filterLineType, entity);
+
+    this.setState(this.getValue());
```

Comment:
This is very dirty. Store the value retrieved from `this.getValue()` in a local variable, call `toggleFilterLineActive(...)` on it and then store this value via `this.setState(localVariable)`